test/integration/test_forward.rb

```diff
@@ -644,4 +656,69 @@ def test_forward_local_unix_socket_to_remote_port
       end
     end
   end
+
+  def test_forward_local_unix_socket_to_remote_socket
+    setup_ssh_env do
+      start_sshd do |_pid, port|
+        session = Timeout.timeout(4) do
```

Comment:
can we put this wait for port to live to `start_sshd`. We don't even have to connect with netssh just wait for the port to be alive until we can make a connection with `TCPSocket.new('localhost', port)`